nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/UnpackContent.java

```diff
@@ -437,6 +463,11 @@ public void unpack(final ProcessSession session, final FlowFile source, final Li
             }
         }
 
+        private record ZipInputStreamMetadata(boolean directory, String zipEntryName, EncryptionMethod encryptionMethod,
+                                               Instant creationTime, Instant lastModifiedDate, Instant lastAccessDate, int mode,
+                                               long uncompressedSize) {
+        }
```

Comment:
Recommend renaming this `record` to `ZipEntryMetadata` as it only applies to a single entry. Also recommend listing each property on its own line for clarity.